### PR TITLE
nextcloud-client: update to 3.1.2, no shibboleth and webkit support, kwallet subpackage

### PR DESCRIPTION
**Pull request by yopito**
* shibboleth is "highly deprecated" by upstream, will be removed in 3.2.
* webkit support is currently ineffective and difficult to fix.
* fix ~dolphin build options
* Edit: add package `nextcloud-client-kwallet` to deal with https://github.com/void-linux/void-packages/issues/26774 . 

#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by ericonr** [comment]
I think we can avoid adding this to `removed-packages` for armv7l if we can move the builders to i686.

**Comment by paper42** [comment]
@yopito could you fix the typo in the commit message? wekbit? 😉

**Comment by yopito** [comment]
> @yopito could you fix the typo in the commit message? wekbit? wink

nice catch ! yes of course.

**Comment by ericonr** [comment]
I don't love this solution, but that's mainly because I don't like vendoring in desktop files. If you can generate the desktop file reliably with a sed on the original desktop file at build time, I'd rather have that.

Besides that, LGTM. Maybe make the INSTALL msg a `README.voidlinux` instead? Otherwise the message will always be there.

**Comment by yopito** [comment]
here it is. looks like this: 
image

**Comment by yopito** [comment]
squashed commits, removed INSTALL.msg, add a README.voidlinux to new subpackage `nextcloud-client-kwallet` .

File: srcpkgs/nextcloud-client/files/kwallet/README.voidlinux [hunk 1]
```diff
@@ -0,0 +1,3 @@
+nextcloud-client-kwallet package provides a workaround to make nextcloud-client
+uses kwallet as its credential storage via qtkeychain-qt5.
+This is usefull when not using KDE as Desktop Environment.
```
Comment (fosslinux) [inline srcpkgs/nextcloud-client/files/kwallet/README.voidlinux:3]: usefull typo

**Review by fosslinux — CHANGES_REQUESTED** [review]

**Comment by q66** [comment]
? why was the `webengine` option removed? the patch applies perfectly fine and doesn't need any rebasing

**Comment by yopito** [comment]
@q66: I don't remember exactly the details, but webkit support is completely useless according to my runtime tests.

**Comment by q66** [comment]
... the webkit stuff and the webengine removal patch are two completely different and separate things

webengine is used for login flow, which the patch disables, it does not use webkit for anything

**Comment by yopito** [comment]
I agree. but without the login flow, can this app be considered as usable ?
I don't see how to use it then.

**Pull request merged**